### ISSUE TITLE
llama_get_embeddings_ith returns NaN/zeros for output slots 2+ with embd-batch decode on Qwen3-VL

### ISSUE THREAD

**Issue by smellslikeml** [ggml-org/llama.cpp#26142]
## Summary

Multi-output `llama_decode` with an **embd batch** on Qwen3-VL models returns undefined content (NaN or uninitialized zeros) for output slots beyond the first 2, regardless of `n_seq_max`. The **token-batch** equivalent with identical mark pattern returns all-valid embeddings. Same-family Qwen2.5 models work correctly on the embd path. Reproduces on the official `Qwen/Qwen3-VL-2B-Instruct-GGUF` and on downstream fine-tunes.

## Environment

- llama.cpp commit: `1425386fd996511e1f3295e7366c38289a92a271`
- Platform: Ubuntu 22.04, x86_64, g++ 11.4.0
- Build flags: default CPU-only, `GGML_CUDA=OFF -DGGML_METAL=OFF -DGGML_VULKAN=OFF -DGGML_BLAS=OFF`
- Model: `Qwen/Qwen3-VL-2B-Instruct-GGUF` (`Qwen3VL-2B-Instruct-Q4_K_M.gguf`, ~1.1 GB)

## Behavior matrix

Same 8-position batch, all marked as outputs (`logits[i] = 1`), context has `embeddings=true, pooling_type=LLAMA_POOLING_TYPE_NONE`:

| Batch type | `n_seq_max` | Slot 0 | Slot 1 | Slot 2 | Slots 3-7 |
|---|---|---|---|---|---|
| **token** | 2 | valid | valid | valid | **valid ✓** |
| **embd** | 2 | valid | valid | zeros (norm=0) | NaN |
| **embd** | 4 | valid | valid | zeros | NaN + zeros mixed |
| **embd** | 8 | valid | valid | zeros | zeros |

Token batches are unaffected by `n_seq_max`. Embd batches return correct output only for slots 0-1; the rest are zeros or NaN.

## Minimal repro

Standalone C file, ~130 LOC, no dependencies beyond llama.cpp itself:

```c
// nseq_output_cap_repro.c
//
// Build:
//   g++ -O2 -x c++ -c nseq_output_cap_repro.c -o repro.o \
//       -I<llama.cpp>/include -I<llama.cpp>/ggml/include
//   g++ -O2 repro.o -o repro \
//       <llama.cpp>/build/src/libllama.a \
//       -Wl,--whole-archive <llama.cpp>/build/ggml/src/libggml-cpu.a -Wl,--no-whole-archive \
//       <llama.cpp>/build/ggml/src/libggml.a \
//       <llama.cpp>/build/ggml/src/libggml-base.a \
//       -lm -ldl -lpthread -fopenmp
//
// Usage:
//   ./repro <model.gguf> <n_seq_max> [--embd] [--seq-cp] [--prior N]

#include "llama.h"
#include <math.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, char **argv) {
    if (argc < 3) {
        fprintf(stderr, "usage: %s <model.gguf> <n_seq_max> [--embd] [--seq-cp] [--prior N]\n", argv[0]);
        return 1;
    }
    const char *model_path = argv[1];
    const int   n_seq_max  = atoi(argv[2]);
    int use_embd = 0, use_seq_cp = 0, n_prior = 0;
    for (int i = 3; i < argc; i++) {
        if      (!strcmp(argv[i], "--embd"))   use_embd = 1;
        else if (!strcmp(argv[i], "--seq-cp")) use_seq_cp = 1;
        else if (!strcmp(argv[i], "--prior") && i + 1 < argc) n_prior = atoi(argv[++i]);
    }
    const int N = 8;

    llama_backend_init();
    llama_model_params mp = llama_model_default_params();
    mp.n_gpu_layers = 0;
    llama_model *model = llama_model_load_from_file(model_path, mp);
    if (!model) { fprintf(stderr, "model load failed\n"); return 2; }

    llama_context_params cp = llama_context_default_params();
    cp.n_ctx           = 256;
    cp.n_batch         = 128;
    cp.n_ubatch        = 128;
    cp.n_seq_max       = n_seq_max;
    cp.embeddings      = true;
    cp.pooling_type    = LLAMA_POOLING_TYPE_NONE;
    cp.no_perf         = true;
    cp.n_threads       = 4;
    cp.n_threads_batch = 4;
    llama_context *ctx = llama_init_from_model(model, cp);
    if (!ctx) { fprintf(stderr, "ctx init failed\n"); return 3; }
    const int n_embd = llama_model_n_embd(model);

    if (n_prior > 0) {
        llama_batch pb = llama_batch_init(1, 0, 1);
        pb.n_tokens = 1;
        pb.n_seq_id[0] = 1;
        pb.seq_id[0][0] = 0;
        pb.logits[0] = 1;
        for (int i = 0; i < n_prior; i++) {
            pb.token[0] = 100 + i;
            pb.pos[0] = i;
            if (llama_decode(ctx, pb) < 0) { fprintf(stderr, "prior decode %d failed\n", i); return 4; }
        }
        llama_batch_free(pb);
    }

    const int target_seq = use_seq_cp ? 1 : 0;
    if (use_seq_cp) {
        llama_memory_t mem = llama_get_memory(ctx);
        llama_memory_seq_rm(mem, 1, 0, -1);
        llama_memory_seq_cp(mem, 0, 1, 0, -1);
    }

    llama_batch b = llama_batch_init(N, use_embd ? n_embd : 0, 1);
    b.n_tokens = N;
    for (int i = 0; i < N; i++) {
        b.pos[i]       = n_prior + i;
        b.n_seq_id[i]  = 1;
        b.seq_id[i][0] = target_seq;
        b.logits[i]    = 1;
        if (use_embd) {
            for (int k = 0; k < n_embd; k++)
                b.embd[i * n_embd + k] = 0.01f * (float)((i * 3 + k) % 7);
        } else {
            b.token[i] = 100 + i;
        }
    }

    if (llama_decode(ctx, b) < 0) { fprintf(stderr, "decode failed\n"); return 5; }

    printf("model n_embd=%d, n_seq_max=%d, batch=%s\n",
           n_embd, n_seq_max, use_embd ? "embd" : "token");
    int n_valid = 0;
    for (int i = 0; i < N; i++) {
        const float *e = llama_get_embeddings_ith(ctx, i);
        if (!e) { printf("  slot %d: NULL\n", i); continue; }
        int has_nan = 0;
        double sum_sq = 0.0;
        for (int k = 0; k < n_embd; k++) {
            if (isnan(e[k])) { has_nan = 1; break; }
            sum_sq += (double)e[k] * (double)e[k];
        }
        if (has_nan) printf("  slot %d: NaN\n", i);
        else { printf("  slot %d: valid, norm=%.2f\n", i, sqrt(sum_sq)); n_valid++; }
    }
    printf("%d/%d slots valid\n", n_valid, N);

    llama_batch_free(b);
    llama_free(ctx);
    llama_model_free(model);
    llama_backend_free();
    return 0;
}
```

## Repro steps

```bash
# 1. Download the model
python3 -c "from huggingface_hub import hf_hub_download; \
  hf_hub_download('Qwen/Qwen3-VL-2B-Instruct-GGUF', \
    'Qwen3VL-2B-Instruct-Q4_K_M.gguf', local_dir='/tmp/qwen3vl')"

# 2. Build the repro (see comment header in the .c file)

# 3. Control (passes):
./repro /tmp/qwen3vl/Qwen3VL-2B-Instruct-Q4_K_M.gguf 2
# → 8/8 slots valid

# 4. Bug (fails):
./repro /tmp/qwen3vl/Qwen3VL-2B-Instruct-Q4_K_M.gguf 2 --embd
# → 3/8 slots valid  (slot 2 = zeros, slots 3-7 = NaN)

./repro /tmp/qwen3vl/Qwen3VL-2B-Instruct-Q4_K_M.gguf 8 --embd
# → 8/8 "valid" but slots 2-7 have norm=0 (zeros, not real computation)
```

## Model specificity

Tested on:

| Model | Family | Embd path result |
|---|---|---|
| `Qwen/Qwen3-VL-2B-Instruct-GGUF` Q4_K_M | Qwen3-VL | **BUG** — 2/8 valid |
| Qwen2.5-family 1.5B (unrelated project's fine-tune) | Qwen2.5 | works — 8/8 valid |

Suggests the bug is architecture-specific to Qwen3-VL (possibly related to its M-RoPE / multi-modal position handling), not a generic embd-path bug.

## Workaround (for anyone who hits this before it's fixed)

Split an N-output decode into ⌈N/2⌉ iterations of exactly-2-output batches. Each iteration sends the full N-position batch (preserving causal attention), but marks only the 2 positions being read. `llama_get_embeddings_ith(0..1)` returns valid values for those 2 marks. Cost: N/2× wall clock.

## Impact

Blocks any use case that needs to read multiple hidden states from a single embd-batch decode on Qwen3-VL — including the InstructSAM mask-query injection pattern (10 hypotheses per phrase), and likely other multi-hypothesis mechanisms that inject synthetic embeddings.

Happy to help narrow this further if a specific ubatch or output-buffer allocation path looks promising.
